pkg/azclient/client-gen/generator/template.go

```diff
@@ -226,36 +230,43 @@ var resourceGroupName = "aks-cit-{{$resource}}"
 var resourceName = "testResource"
 {{if .SubResource}}var {{toLower .Resource}}Name = "testParentResource"{{- end }}
 var subscriptionID string
-var location = "eastus"
+var location string
 var resourceGroupClient *armresources.ResourceGroupsClient
 var err error
 var recorder *recording.Recorder
+var cloudConfig *cloud.Configuration
 var realClient Interface
+var clientOption azcore.ClientOptions 
 
 var _ = ginkgo.BeforeSuite(func(ctx context.Context) {
-	recorder, err = recording.NewRecorder("testdata/{{$resource}}")
+	recorder, cloudConfig, location, err = recording.NewRecorder()
 	gomega.Expect(err).ToNot(gomega.HaveOccurred())
 	subscriptionID = recorder.SubscriptionID()
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	cred := recorder.TokenCredential()
-	resourceGroupClient, err = armresources.NewResourceGroupsClient(subscriptionID, cred, &arm.ClientOptions{
-		ClientOptions: azcore.ClientOptions{
-			Transport: recorder.HTTPClient(),
-			TracingProvider: utils.TracingProvider,
-			Telemetry: policy.TelemetryOptions{
-				ApplicationID: "ccm-resource-group-client",
-			},
+	clientOption = azcore.ClientOptions{
+		Transport:       recorder.HTTPClient(),
+		TracingProvider: utils.TracingProvider,
+		Telemetry: policy.TelemetryOptions{
+			ApplicationID: "ccm-resource-group-client",
 		},
+		Cloud: *cloudConfig,
+	}
+	rgClientOption := clientOption
+	rgClientOption.Telemetry.ApplicationID = "ccm-resource-group-client"
+	resourceGroupClient, err = armresources.NewResourceGroupsClient(subscriptionID, cred, &arm.ClientOptions{
+		ClientOptions: rgClientOption,
 	})
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	realClientOption := clientOption
+	realClientOption.Telemetry.ApplicationID = "ccm-{{$resource}}-client"
+{{- if .MooncakeApiVersion}}
+	if location == "chinaeast2" {
```

Comment:
this won't work in other azure china regions and other sovereign regions, let's use `2023-05-01` for all public regions now.

Reply:
and requests recorded in yaml. we are using 2023-05-01 in account/blob/fileshare clients.